blockchain/escrow.go

```diff
@@ -0,0 +1,59 @@
+package blockchain
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	log "github.com/sirupsen/logrus"
+)
+
+func (processor *Processor) ClaimFundsFromChannel(timeout time.Duration, channelId, amount *big.Int, signature []byte, sendBack bool) (err error) {
+	log := log.WithFields(log.Fields{
+		"timeout":    timeout,
+		"channelId":  channelId,
+		"amount":     amount,
+		"signature":  BytesToBase64(signature),
+		"isSendBack": sendBack,
+	})
+
+	auth := bind.NewKeyedTransactor(processor.privateKey)
+
+	log.Info("Submitting transaction to claim funds from channel")
+	txn, err := processor.multiPartyEscrow.ChannelClaim(
+		&bind.TransactOpts{
+			From:     common.HexToAddress(processor.address),
+			Signer:   auth.Signer,
+			GasLimit: 1000000,
+		},
+		channelId,
+		amount,
+		signature,
+		sendBack,
+	)
+	if err != nil {
+		log.WithError(err).Error("Error submitting transaction to claim funds from channel")
+		return fmt.Errorf("Error submitting transaction to claim funds from channel: %v", err)
+	}
+
+	log.Info("Transaction sent, waiting for %v till transaction is committed", timeout)
+	endTime := time.Now().Add(timeout)
+	isPending := true
+	for {
+		_, isPending, _ = processor.ethClient.TransactionByHash(context.Background(), txn.Hash())
```

Comment:
It is also possible  that isPending is false but returns error is not nil.
In this case the transaction should not be treated as successful.